SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -250,143 +179,36 @@ bool FLocalDeploymentManager::LocalDeploymentPreRunChecks()
 {
 	bool bSuccess = true;
 
-	// Check for the known runtime port (5301) which could be blocked.
-	if (CheckIfPortIsBound(RequiredRuntimePort))
-	{
-		// If it exists offer the user the ability to kill it.
-		if (FMessageDialog::Open(EAppMsgType::YesNo, LOCTEXT("KillPortBlockingProcess",
-															 "A required port is blocked by another process (potentially by an old "
-															 "deployment). Would you like to kill this process?"))
-			== EAppReturnType::Yes)
-		{
-			bSuccess = KillProcessBlockingPort(RequiredRuntimePort);
-		}
-		else
-		{
-			bSuccess = false;
-		}
-	}
+	// Check for the known runtime ports which could be blocked by other processes.
+	TArray<int32> RequiredRuntimePorts = { RequiredRuntimePort, WorkerPort, HTTPPort, GRPCPort };
 
-	if (!bSpatialServiceInProjectDirectory && bSpatialServiceRunning)
+	for (int32 RuntimePort : RequiredRuntimePorts)
 	{
-		if (FMessageDialog::Open(EAppMsgType::YesNo, LOCTEXT("StopSpatialServiceFromDifferentProject",
-															 "An instance of the SpatialOS Runtime is running with another project. Would "
-															 "you like to stop it and start the Runtime for this project?"))
-			== EAppReturnType::Yes)
+		if (CheckIfPortIsBound(RequiredRuntimePort))
 		{
-			bSuccess = TryStopSpatialService();
-		}
-		else
-		{
-			bSuccess = false;
+			// If it exists offer the user the ability to kill it.
+			if (FMessageDialog::Open(EAppMsgType::YesNo, LOCTEXT("KillPortBlockingProcess",
```

Comment:
I would prefer this not to be directly in the if statement. it's just too long xD

Reply:
Prefer what to not be in the statement? The dialog that opens? What would you write instead. This is based on the existing code:
```
		if (FMessageDialog::Open(EAppMsgType::YesNo, LOCTEXT("StopSpatialServiceFromDifferentProject",
															 "An instance of the SpatialOS Runtime is running with another project. Would "
															 "you like to stop it and start the Runtime for this project?"))
			== EAppReturnType::Yes)
		{
			bSuccess = TryStopSpatialService();
		}
```

Reply:
You could store the LOCTEXT in a local variable above the if statement?

Reply:
Right I understand now! Done